tests/test_scalars.py

```diff
@@ -1931,128 +1825,144 @@ def test_period_add_subtract() -> None:
+
+    # >, <=
+    gt1 = check(assert_type(p > p_ser, "pd.Series[bool]"), pd.Series, np.bool)
+    le1 = check(assert_type(p <= p_ser, "pd.Series[bool]"), pd.Series, np.bool)
+    assert (gt1 != le1).all()
+    gt2 = check(assert_type(p_ser > p, "pd.Series[bool]"), pd.Series, np.bool)
+    le2 = check(assert_type(p_ser <= p, "pd.Series[bool]"), pd.Series, np.bool)
+    assert (gt2 != le2).all()
+
+    # <, >=
+    lt1 = check(assert_type(p < p_ser, "pd.Series[bool]"), pd.Series, np.bool)
+    ge1 = check(assert_type(p >= p_ser, "pd.Series[bool]"), pd.Series, np.bool)
+    assert (lt1 != ge1).all()
+    lt2 = check(assert_type(p_ser < p, "pd.Series[bool]"), pd.Series, np.bool)
+    ge2 = check(assert_type(p_ser >= p, "pd.Series[bool]"), pd.Series, np.bool)
+    assert (lt2 != ge2).all()
+
+    # ==, !=
+    eq1 = check(assert_type(p == p_ser, "pd.Series[bool]"), pd.Series, np.bool)
+    ne1 = check(assert_type(p != p_ser, "pd.Series[bool]"), pd.Series, np.bool)
+    assert (eq1 != ne1).all()
+
+    # ==, != (p on the rhs, use == and != of lhs)
+    eq_rhs1 = check(assert_type(p_ser == p, "pd.Series[bool]"), pd.Series, np.bool)
+    ne_rhs1 = check(assert_type(p_ser != p, "pd.Series[bool]"), pd.Series, np.bool)
+    assert (eq_rhs1 != ne_rhs1).all()
+
+
+def test_period_cmp_index() -> None:
+    p = pd.Period("2012-1-1", freq="D")
+    p_idx = pd.period_range("2012-1-1", periods=10, freq="D")
+
+    # >, <=
+    gt1 = check(assert_type(p > p_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    le1 = check(assert_type(p <= p_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (gt1 != le1).all()
+    gt2 = check(assert_type(p_idx > p, np_1darray[np.bool]), np_1darray[np.bool])
+    le2 = check(assert_type(p_idx <= p, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (gt2 != le2).all()
+
+    # <, >=
+    lt1 = check(assert_type(p < p_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    ge1 = check(assert_type(p >= p_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (lt1 != ge1).all()
+    lt2 = check(assert_type(p_idx < p, np_1darray[np.bool]), np_1darray[np.bool])
+    ge2 = check(assert_type(p_idx >= p, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (lt2 != ge2).all()
+
+    # ==, !=
+    eq1 = check(assert_type(p == p_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    ne1 = check(assert_type(p != p_idx, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (eq1 != ne1).all()
+
+    # ==, != (p on the rhs, use == and != of lhs)
+    eq_rhs1 = check(assert_type(p_idx == p, np_1darray[np.bool]), np_1darray[np.bool])
+    ne_rhs1 = check(assert_type(p_idx != p, np_1darray[np.bool]), np_1darray[np.bool])
+    assert (eq_rhs1 != ne_rhs1).all()
+
+
+def test_period_cmp_array() -> None:
+    p = pd.Period("2012-1-1", freq="D")
+    arr_nd = cast(  # cast to avoid pyright narrowing to a more precise shape
+        npt.NDArray[np.object_],
+        pd.period_range("2012-1-1", periods=4, freq="D").to_numpy(),
+    )
```

Comment:
So the reason that is happening is because we know that `period_range()` returns a `PeriodIndex` and `PeriodIndex` returns a 1D numpy array.  

So I think you should add a test for the actual result of `to_numpy()`, i.e., you can test the non-precise shape version, and the version that is precise.